util/topgen/resets.py

```diff
@@ -0,0 +1,181 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, Optional
+from .clocks import Clocks
+
+
+class ResetItem:
+    '''Individual resets'''
+    def __init__(self, hier: Dict[str, str], raw: Dict[str, object], clocks: Clocks):
+        if not raw['name']:
+            raise ValueError('Reset has no name')
+
+        self.name = raw['name']
+        self.gen = raw.get('gen', True)
+        self.rst_type = raw.get('type', 'top')
+
+        self.path = ""
+        if self.rst_type == 'top':
+            self.path = f"{hier['top']}rst_{self.name}_n"
+        elif self.rst_type == 'ext':
+            self.path = f"{hier['ext']}{self.name}"
+
+        # to be constructed later
+        self.domains = []
+        self.shadowed = False
+
+        self.parent = raw.get('parent', "")
+
+        # This can be a source clock or a derived source
+        if self.rst_type != 'ext':
+            self.clock = clocks.get_clock_by_name(raw['clk'])
+        else:
+            self.clock = None
+
+        self.sw = bool(raw.get('sw', 0))
+
+    def _asdict(self) -> Dict[str, object]:
+        ret = {
+            'name': self.name,
+            'gen': self.gen,
+            'type': self.rst_type,
+            'domains': self.domains,
+            'shadowed': self.shadowed,
+            'parent': self.parent,
```

Comment:
If you want to avoid adding bogus empty-string parents in the generated hjson, I suggest changing line 29 to `self.parent = raw.get('parent')` (so it has value `None` if the key wasn't there) and then removing this line and adding
```python3
        if self.parent is not None:
            ret['parent'] = self.parent
```
below.